thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/wrapper/ChildKeepingMergeWrapper.java

```diff
@@ -91,10 +93,13 @@ public ChildKeepingMergeWrapper(DataProvider provider, DetectionConfigDTO config
       if (anomaly.getProperties().containsKey(PROP_GROUP_KEY)) {
         groupKey = anomaly.getProperties().get(PROP_GROUP_KEY);
       }
-
+      String patternKey = "";
+      if (anomaly.getProperties().containsKey(PROP_PATTERN_KEY)) {
+        patternKey = anomaly.getProperties().get(PROP_PATTERN_KEY);
```

Comment:
some anomalies don't have the pattern key. in that case, should it be deducted from the current and the baseline?

Reply:
I think that a few rule-based detectors don't fill out the baseline value of `MergedAnomalyResult`, `ThresholdRuleDetector` for instance, so comparing them may not work for all detectors either.